components/DataTable/DataTable.tsx

```diff
@@ -28,7 +29,8 @@ interface DataTableProps<TData, TValue> {
   data: TData[];
   filterColumnAccessorKey?: string;
   handleDeleteSelected?: (data: TData[]) => Promise<void> | void;
-  actions?: ColumnDef<TData, TValue>[];
+  actions?: React.ComponentType<{ row: Row<TData>; data: TData[] }>;
```

Comment:
Nice! Didn't know this is how you did it. This is super cool, because it also validates the data type being passed to the table!